File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    let queue = [];
    queue.push(root);
    
    let output = [];

    while(queue[0]) {
        const qLength = queue.length;
        let row = [];

        for (let i = 0; i < qLength; i++) {
            const curr = queue.shift();
            row.push(curr.val)
            if(curr.left) queue.push(curr.left);
            if(curr.right) queue.push(curr.right);

        }

        if(row) {
            output.push(row)
        }
    }
    return output;
};

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root: return []
        queue = deque()
        queue.append(root)
        res = []

        while len(queue) > 0:
            rowLength = len(queue)
            row = []

            for i in range(rowLength):
                curr = queue.popleft()
                row.append(curr.val)

                if (curr.left): queue.append(curr.left)
                if (curr.right): queue.append(curr.right)

            res.append(row)
        return res
